core/wiki/macros/tag-picker.tid

```diff
@@ -28,73 +28,125 @@ $actions$
-<$vars storeTitle={{{ [<newTagNameInputTiddler>!match[]] ~[<newTagNameInputTiddlerQualified>] }}} tagSelectionState={{{ [<newTagNameSelectionTiddler>!match[]] ~[<newTagNameSelectionTiddlerQualified>] }}}>
-<$vars refreshTitle=<<qualify "$:/temp/NewTagName/refresh">> nonSystemTagsFilter="[tags[]!is[system]search:title<userInput>sort[]]" systemTagsFilter="[tags[]is[system]search:title<userInput>sort[]]">
+<$let 
+newTagNameInputTiddlerQualified=<<qualify "$:/temp/NewTagName/input">> 
+newTagNameSelectionTiddlerQualified=<<qualify "$:/temp/NewTagName/selected-item">> 
+fallbackTarget={{$(palette)$##tag-background}}
+colourA={{$(palette)$##foreground}}
+colourB={{$(palette)$##background}}
+storeTitle={{{ [<newTagNameInputTiddler>!match[]] ~[<newTagNameInputTiddlerQualified>] }}} 
+tagSelectionState={{{ [<newTagNameSelectionTiddler>!match[]] ~[<newTagNameSelectionTiddlerQualified>] }}}
+refreshTitle=<<qualify "$:/temp/NewTagName/refresh">>
+nonSystemTagsFilter="[tags[]] $(first-search-filter)$ +[!is[system]] -[<storyTiddler>tags[]] :filter[search:title<userInput>]+[sort[]]"
+systemTagsFilter="[tags[]] $(second-search-filter)$ +[is[system]] -[<storyTiddler>tags[]] :filter[search:title<userInput>]+[sort[]]"
+displayTagsPopup="[all[tiddlers]subfilter<systemTagsFilter>][all[tiddlers]subfilter<nonSystemTagsFilter>] +[limit[1]]"
+>
 <div class="tc-edit-add-tag">
 <div>
 <span class="tc-add-tag-name tc-small-gap-right">
-<$macrocall $name="keyboard-driven-input" tiddler=<<newTagNameTiddler>> storeTitle=<<storeTitle>> refreshTitle=<<refreshTitle>>
-		selectionStateTitle=<<tagSelectionState>> inputAcceptActions="<$macrocall $name='add-tag-actions' actions=<<__actions__>> tagField=<<__tagField__>>/>"
-		inputCancelActions=<<clear-tags-actions>> tag="input" placeholder={{$:/language/EditTemplate/Tags/Add/Placeholder}}
-		focusPopup=<<qualify "$:/state/popup/tags-auto-complete">> class="tc-edit-texteditor tc-popup-handle" tabindex=<<tabIndex>> 
-		focus={{{ [{$:/config/AutoFocus}match[tags]then[true]] ~[[false]] }}} filterMinLength={{$:/config/Tags/MinLength}} 
-		cancelPopups=<<cancelPopups>> configTiddlerFilter="[[$:/core/macros/tag-picker]]"/>
-</span><$button popup=<<qualify "$:/state/popup/tags-auto-complete">> class="tc-btn-invisible tc-btn-dropdown" tooltip={{$:/language/EditTemplate/Tags/Dropdown/Hint}} aria-label={{$:/language/EditTemplate/Tags/Dropdown/Caption}}>{{$:/core/images/down-arrow}}</$button><$reveal state=<<storeTitle>> type="nomatch" text=""><$button class="tc-btn-invisible tc-small-gap tc-btn-dropdown" tooltip={{$:/language/EditTemplate/Tags/ClearInput/Hint}} aria-label={{$:/language/EditTemplate/Tags/ClearInput/Caption}}>{{$:/core/images/close-button}}<<delete-tag-state-tiddlers>></$button></$reveal><span class="tc-add-tag-button tc-small-gap-left">
-<$set name="tag" value={{{ [<newTagNameTiddler>get[text]] }}}>
-<$button set=<<newTagNameTiddler>> setTo="" class="">
+<$macrocall $name="keyboard-driven-input"
```

Comment:
I personally don't like macrocalls, that create 1 line for every parameter. I think they don't improve readability at all. ... But it's just me. 

May be indentation will help

Reply:
I originally indented my code but I noticed that there was none on the tiddler so I removed the indents, I will add them then 👌

Reply:
> As you wrote above, I expect no popup for empty or no result
> 
> I assume the mechanism for displaying dropdown is to see if the input is focused or non-empty, while it can be changed to: show dropdown if there is any result. I think this is more semantic, but one may believe to show a message instead of above empty dropdown with horizontal line!

@kookma the popup should be fixed now. For the focus indication there is the blinking cursor, and the search doesnt show a popup when nothing is written so in my opinion it's a bit more consistent to have the same behavior in the tag picker, but I can revert this change if needed

Reply:
Hum seems like I made a mistake, I'm working on it